x/supplier/keeper/msg_server_unstake_supplier.go

```diff
@@ -36,6 +36,16 @@ func (k msgServer) UnstakeSupplier(
 		logger.Info(fmt.Sprintf("Supplier not found. Cannot unstake address %s", msg.Address))
 		return nil, types.ErrSupplierNotFound
 	}
+
+	// Ensure that the message sender is the supplier owner.
+	if supplier.OwnerAddress != msg.OwnerAddress {
```

Comment:
We wouldn't have to copy-paste this if it were part of ValidateBasic -> enables unit testing too.

Reply:
This will be `supplier.EnsureOwner(msg.OwnerAddress)`